core/src/serde_support.rs

```diff
@@ -0,0 +1,1458 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::format;
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+use base64 as _;
+use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
+use base64::Engine;
+use dusk_bytes::Serializable;
+use hex as _;
+use serde::de::{
+    EnumAccess, Error as SerdeError, MapAccess, Unexpected, VariantAccess,
+    Visitor,
+};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json as _;
+
+use crate::stake::{Reward, RewardReason, SlashEvent, StakeEvent, StakeKeys};
+use crate::transfer::withdraw::WithdrawReceiver;
+use crate::transfer::{
+    ContractToAccountEvent, ContractToContractEvent, ConvertEvent,
+    DepositEvent, MoonlightTransactionEvent, PhoenixTransactionEvent,
+    WithdrawEvent,
+};
+use crate::{
+    signatures::bls::PublicKey as AccountPublicKey, stake::StakeFundOwner,
+};
+use crate::{BlsScalar, String};
+
+// To serialize and deserialize u64s as big ints:
+// https://github.com/dusk-network/rusk/issues/2773#issuecomment-2519791322.
+struct Bigint(u64);
+
+impl Serialize for Bigint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s: String = format!("{}n", self.0);
+        s.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Bigint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let mut s = String::deserialize(deserializer)?;
+        let last_char = s.pop().ok_or_else(|| {
+            SerdeError::invalid_value(
+                Unexpected::Str(&s),
+                &"a non-empty string",
+            )
+        })?;
+        if last_char != 'n' {
+            return Err(SerdeError::invalid_value(
+                Unexpected::Str(&s),
+                &"a bigint ending with character 'n'",
+            ));
+        }
```

Comment:
I am not quite sure if we want to enforce this on the Deserialization part. Theoretically, we could support both without a problem. I will leave this comment here for further discussion.